src/index.js

```diff
@@ -0,0 +1,344 @@
+    const resWeather = await fetchWeather(resLatLon);
+    state.temp = resWeather.current.temp;
+    const curWeatherDes = resWeather.current.weather[0].description;
+    const curWeatherMain = resWeather.current.weather[0].main;
+    state.timezone = resWeather.timezone;
+    
+    if (curWeatherMain === "Clouds") {
+        skyInput.value = "Cloudy";
+    } else if (curWeatherMain === "Rain") {
+        skyInput.value = "Rainy";
+    } else if (curWeatherMain === "Clear") {
+        skyInput.value = "Sunny";
+    } else {
+        skyInput.value = "Snowy";
+    }
+    state.cur_conditions = curWeatherDes;
+
+    fetchForecast(resWeather.daily);
+
+    displayStates();
+}
+
+const changeUnits = () => {
+    if (state.unit === 'F') {
+        state.unit = 'C';
+        curUnits.textContent = ' °C / '
+        changeUnitsBtn.textContent =  '°F'
+        state.temp = (state.temp-32)*(5/9);
+        for (const dayObj of forecastArr) {
+            dayObj.min = (dayObj.min-32)*(5/9);
+            dayObj.max = (dayObj.max-32)*(5/9);
+        }
+    } else {
+        state.unit = 'F';
+        curUnits.textContent = ' °F / '
+        changeUnitsBtn.textContent =  '°C'
+        state.temp = (state.temp*1.8)+32;
+        for (const dayObj of forecastArr) {
+            dayObj.min = (dayObj.min*1.8)+32;
+            dayObj.max = (dayObj.max*1.8)+32;
+        }
+    }
+    displayStates();
+}
+
+const fetchCity = async (lat, lon) => {
+    try {
+        const response = await axios.get("https://weather-report-proxy.herokuapp.com/location/reverse", {params: {lat: lat, lon: lon}});
+        // const response = await axios.get(`https://weather-report-proxy.herokuapp.com/location/reverse?lat=${lat}&lon=${lon}`);
+        errorBox.innerHTML = "";
+        errorBox.className = "";
+        return response.data.address.city;
+    } catch (e) {
+        errorBox.className = "error";
+        errorBox.textContent = "ERROR: Unable to fetch city from lat/lon";
+    }
+};
+
+const fetchLiveWeather = () => {
+    textInputBox.value = "";
+    const success = async (pos) => {
+        alert('Please wait to fetch your current location.');
+        const curCity = await fetchCity(pos.coords.latitude, pos.coords.longitude);
+        if (curCity) {
+            state.title = curCity;
+        }
+        await fetchAll();
+    };
+    const error = (err) => {
+        alert('Your browser does not permit location sharing.');
+    };
+    navigator.geolocation.getCurrentPosition(success, error);
+};
+
+const registerEventHandlers = () => {
+    increaseTempBtn.addEventListener('click', increaseTemp);
+    decreaseTempBtn.addEventListener('click', decreaseTemp);
+    textInputBox.addEventListener('input', updateTitle);
+    skyInput.addEventListener('change', updateSkyState);
+    fetchWeatherBtn.addEventListener('click', fetchAll);
+    fetchLocationWeatherBtn.addEventListener('click', fetchLiveWeather);
+    changeUnitsBtn.addEventListener('click', changeUnits);
+}
```

Comment:
Small nit: Consider organizing all event handler related code at the bottom of the file, in this case, below L341.